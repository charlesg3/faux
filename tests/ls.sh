#!/bin/bash
source `dirname $0`/checks.sh
start_test

umask 022

make_dir d1
make_dir d2
touch f1
make_file f2 "hello world."
make_file d1/f1 "hello world."
ln -s d1/f1 s1
ln -s /does/not/exist s2

ls -la --time-style=+"" | grep -v total > /$testname.out

files_same /$testname.out $HARE_ROOT/tests/files/ls.expected
remove_file /$testname.out

remove_file d1/f1
remove_file f2
remove_file s1
remove_file s2
remove_dir d1
remove_dir d2
remove_file f1

end_test

