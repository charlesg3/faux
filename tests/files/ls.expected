drwxrwxrwx 0 root root 4096  .
drwxr-xr-x 0 root root 4096  ..
drwxr-xr-x 0 root root 4096  d1
drwxr-xr-x 0 root root 4096  d2
-rw-rw-rw- 0 root root    0  f1
-rw-rw-rw- 0 root root   13  f2
lrwxrwxrwx 0 root root    5  s1 -> d1/f1
lrwxrwxrwx 0 root root   15  s2 -> /does/not/exist
